app/models/terminal_executor.rb

```diff
@@ -131,7 +134,19 @@ def script(commands)
   end
 
   def print_and_execute(c, resolve: true)
-    "echo » #{c.shellescape}\n#{resolve ? resolve_secrets(c) : c}"
+    print_and_execute_hidden_command(c) ||
+      print_and_execute_raw(c, resolve ? resolve_secrets(c) : c)
+  end
+
+  def print_and_execute_hidden_command(c)
+    # hides secrets by replacing lines like export FOO="hidden://secret" into export FOO="HIDDEN"
+    return unless print_command = c.dup.sub!(/#{HIDDEN_PREFIX}[^"]+/, HIDDEN_TXT)
```

Comment:
looking at this again I think it needs to get better eventually since some secrets will include `"`, maybe `hidden://fooo//`, but as long as it's only used internally thats good enough for now ...